app/src/main/java/com/teamwss/websoso/ui/postNovel/DatePickerDialog.kt

```diff
@@ -0,0 +1,285 @@
+            npPostDatePickerMonth.setFormatter { String.format("%02d", it) }
+            npPostDatePickerDay.setFormatter { String.format("%02d", it) }
+        }
+    }
+
+    // 다이얼로그 설정
+    private fun setupDialog() {
+        // 배경 투명화를 통해 둥근 다이얼로그
+        window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+        window?.attributes?.windowAnimations = R.style.DialogAnimation
+        window?.setGravity(Gravity.BOTTOM)
+        window?.setLayout(
+            ViewGroup.LayoutParams.MATCH_PARENT,
+            ViewGroup.LayoutParams.WRAP_CONTENT
+        )
+    }
+
+    // 월 체크
+    private fun setDayMaxValue() {
+        val year = binding.npPostDatePickerYear.value
+        val month = binding.npPostDatePickerMonth.value
+
+        // 월에 따른 일의 최대값 설정
+        binding.npPostDatePickerDay.maxValue = when (month) {
+            2 -> if (isLeapYear(year)) 29 else 28
+            4, 6, 9, 11 -> 30
+            else -> 31
+        }
+    }
+
+    // 윤년 체크
+    private fun isLeapYear(year: Int): Boolean {
+        return (year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)
+    }
+
+    // 완료 버튼 클릭 이벤트 설정
+    private fun setupPostButtonClickListener() {
+        binding.llDatePostButton.setOnClickListener {
+            setupAnotherDateValid()
+            inputSelectedDateListener.inputSelectedDate(startDate, endDate)
+            dismiss()
+        }
+    }
+
+    // selector 클릭 이벤트 설정
+    private fun setupDateTypeSelector() {
+        with(binding) {
+            updateDateSelector(true)
+            updateDateNumberPicker(true)
+
+            llPostDatePickerReadDateStart.setOnClickListener {
+                updateDateSelector(true)
+                updateDateNumberPicker(true)
+            }
+
+            llPostDatePickerReadDateEnd.setOnClickListener {
+                updateDateSelector(false)
+                updateDateNumberPicker(false)
+            }
+        }
+    }
+
+    // selector 업데이트
+    private fun updateDateSelector(isStart: Boolean) {
+        with(binding) {
+            llPostDatePickerReadDateStart.isSelected = isStart
+            llPostDatePickerReadDateEnd.isSelected = !isStart
+
+            val startColor = if (isStart) R.color.primary_100_6341F0 else R.color.gray_100_CBCBD1
+            val endColor = if (!isStart) R.color.primary_100_6341F0 else R.color.gray_100_CBCBD1
+
+            tvPostDatePickerReadDateStartTitle.setTextColor(
+                ContextCompat.getColor(
+                    context,
+                    startColor
+                )
+            )
+            tvPostDatePickerReadDateStart.setTextColor(ContextCompat.getColor(context, startColor))
+
+            tvPostDatePickerReadDateEndTitle.setTextColor(
+                ContextCompat.getColor(
+                    context,
+                    endColor
+                )
+            )
+            tvPostDatePickerReadDateEnd.setTextColor(ContextCompat.getColor(context, endColor))
```

Comment:
이렇게 하는 것은 어떻게 생각하시나요
```suggestion
val startColorResId = if (isStart) R.color.primary_100_6341F0 else R.color.gray_100_CBCBD1
val endColorResId = if (!isStart) R.color.primary_100_6341F0 else R.color.gray_100_CBCBD1

val startColorValue = ContextCompat.getColor(context, startColorResId)
val endColorValue = ContextCompat.getColor(context, endColorResId)

tvPostDatePickerReadDateStartTitle.setTextColor(startColorValue)
tvPostDatePickerReadDateStart.setTextColor(startColorValue)
tvPostDatePickerReadDateEndTitle.setTextColor(endColorValue)
tvPostDatePickerReadDateEnd.setTextColor(endColorValue)
```